x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/RestGetShutdownStatusAction.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.shutdown;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+
+import java.util.List;
+
+public class RestGetShutdownStatusAction extends BaseRestHandler {
+
+    @Override
+    public String getName() {
+        return "get_shutdown_status";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(RestRequest.Method.GET, "/_node/{name}/shutdown"), new Route(RestRequest.Method.GET, "/_node/shutdown"));
```

Comment:
nit: doesn't `/_node/{name}/shutdown` conflict with `/_node/shutdown` since you cannot have a node named `shutdown`?

Reply:
No, we do this all over the place already, for example, in `RestNodesStatsAction` we have:

```java
            new Route(GET, "/_nodes/stats"),
            new Route(GET, "/_nodes/{nodeId}/stats"),
```